### PR TITLE
style: fix InputNumber prefix style in Form.Item

### LINKED ISSUE #32924: InputNumber does not handle currency/decimals properly
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[Edit on CodeSandbox](https://codesandbox.io/s/currency-wrapper-antd-input-3ynzo)

### Steps to reproduce
(Notice, linked code sandbox is listed as an example on official documentation)
1. Choose USD. If there is a value, highlight the entire field by double clicking. 
2. Try to type in "10.25"
3. Notice "$10.00.2" appears, then on blur, we are left with "10.00"
4. There is some issue when user types 0 or tries to add decimals. 

### What is expected?
User should be able to type in a currency amount with decimals properly. In this case, user can't type after the decimal properly.  It seems the component is re-rendering way more often than the 3.26.17 version.  In the previous version, the component only re-rendered & reformatted the result after the user blurred the input. Now, the component is updating the value with every keystroke, making it very difficult to handle decimals -- even for the given example application. It should instead only re-format after the user blurs.

### What is actually happening?
There is some issue when user types 0. The elements re-renders too often; It should really only update the decimal after the user blurs.

| Environment | Info |
|---|---|
| antd | 4.17.0 |
| React | ^16.13.1, latest |
| System | Mac/Windows, * |
| Browser | Chrome, * |

---
After updating from ^3.26.17, we found our Input formatter no longer functioned properly for prices. I came here to see documentation on the update, and found even the antd example contains the same problem.

### LINKED ISSUE #35249: Style of InputNumber in Input.Group compact mode is broken.
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/inputnumber-style-bug-under-inputgroup-under-feedback-enabled-form-cxgijx?file=/demo.js)

### Steps to reproduce

See [demo](https://codesandbox.io/s/inputnumber-style-bug-under-inputgroup-under-feedback-enabled-form-cxgijx?file=/demo.js)

### What is expected?

The `<InputNumber />` wrapper div displayed as `inline-flex`

### What is actually happening?

The `<InputNumber />` wrapper div displayed as `inline-block`, so the style is broken.

| Environment | Info |
| --- | --- |
| antd | 4.20.0 |
| React | 17.0.2 |
| System | Pop!_OS 22.04 LTS |
| Browser | MS Edge 100.0.1185.50 |

### PR DESCRIPTION
**Pull request by afc163**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

close #35249

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |   Fix InputNumber with `prefix` abnormal height under Form.Item of `hasFeedBack`.      |
| 🇨🇳 Chinese |  修复带 `prefix` InputNumber 在 Form.Item `hasFeedBack` 内高度异常的问题。         |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### 🔍 Walkthrough

<!--
copilot:walkthrough
--

### CODE REVIEW

File: components/input/style/index.ts [hunk 1]
```diff
@@ -406,7 +406,7 @@ export const genInputGroupStyle = (token: InputToken): CSSObject => {
       },
 
       '& > *': {
-        display: 'inline-block',
+        display: 'inline-flex',
```
Comment (afc163) [inline components/input/style/index.ts:409]:
```suggestion
        display: 'inline-flex',
        alignItems: 'center',
```

File: components/input-number/style/index.ts [hunk 2]
```diff
@@ -359,6 +359,7 @@ const genAffixWrapperStyles: GenerateStyle<InputNumberToken> = (token: InputNumb
       },
 
       '&::before': {
+        display: 'inline-block',
```
Comment (zombieJ) [inline components/input-number/style/index.ts:362]: 这个 before 还有用么？感觉是用来 baseline 对齐的，不过我们现在都推荐 Space 了

File: components/input/style/index.ts [hunk 3]
```diff
@@ -412,11 +412,11 @@ export const genInputGroupStyle = (token: InputToken): CSSObject => {
...
       },
 
-      [`& > ${componentCls}-affix-wrapper`]: {
-        display: 'inline-flex',
-      },
-
-      [`& > ${antCls}-picker-range`]: {
+      [`
+        & > ${componentCls}-affix-wrapper,
+        & > ${componentCls}-number-affix-wrapper,
```
Comment (zombieJ) [inline components/input/style/index.ts:417]: InputNumber 的样式放这里不好，因为如果页面里只有 InputNumber 没有 Input，这个样式是不会加载进来的

Comment (zombieJ) [reply components/input/style/index.ts:417]: 哦，是 Group 样式。那无所谓了

**Review by zombieJ — APPROVED** [review]

File: components/input-number/style/index.ts (hunk 2 above)
Comment (afc163) [reply components/input-number/style/index.ts:362]: 有用，不走 Space 也尽可能能对齐吧，不动了。

**Pull request merged**